app/services/clients/ledger.client.js

```diff
@@ -18,53 +20,50 @@ const defaultOptions = {
   limit_total: true
 }
 
-const transaction = function transaction (id, gatewayAccountId, options = {}) {
-  const configuration = Object.assign({
-    url: `/v1/transaction/${id}`,
-    qs: {
-      account_id: gatewayAccountId,
-      ...options.transaction_type && { transaction_type: options.transaction_type }
-    },
-    description: 'Get individual transaction details',
-    transform: legacyConnectorTransactionParity
-  }, defaultOptions, options)
-  return baseClient.get(configuration)
+const transaction = async function transaction (id, gatewayAccountId, options = {}) {
+  this.client = new Client(defaultOptions.service)
+  const baseUrl = options.baseUrl ? options.baseUrl : defaultOptions.baseUrl
+  let url = `${baseUrl}/v1/transaction/${id}?account_id=${gatewayAccountId}`
+  if ( options.transaction_type ) {
+    url = `${url}&transaction_type=${options.transaction_type}`
+  }
+  configureClient(this.client, url)
+  const response = await this.client.get(url, 'Get individual transaction details')
+  const body = legacyConnectorTransactionParity(response.data)
+  return body
 }
 
-const transactionWithAccountOverride = function transactionWithAccountOverride (id, options = {}) {
-  const configuration = Object.assign({
-    url: `/v1/transaction/${id}`,
-    qs: {
-      override_account_id_restriction: true
-    },
-    description: 'Get individual transaction details with no accountId restriction'
-  }, defaultOptions, options)
-  return baseClient.get(configuration)
+const transactionWithAccountOverride = async function transactionWithAccountOverride (id, options = {}) {
+  const url = urlJoin(defaultOptions.baseUrl,'/v1/transaction', id)
```

Comment:
Do we need to add a check as per line 24?
Check if the baseUrl is the default URL / coming from options object.